packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -15,30 +20,70 @@ import { useEventCallback } from '@fluentui/react-utilities';
  * @param ref - reference to root HTMLElement of TreeItem
  */
-export const useTreeItem_unstable = (props: TreeItemProps, ref: React.Ref<TreeItemElement>): TreeItemState => {
+export const useTreeItem_unstable = (props: TreeItemProps, ref: React.Ref<HTMLDivElement>): TreeItemState => {
   const treeItemState = useBaseTreeItem_unstable(props, ref);
-  const { expandIcon, iconBefore, iconAfter, actions, badges } = props;
-  const { dir } = useFluent_unstable();
+  const { expandIcon, iconBefore, iconAfter, actions, badges, groupper } = props;
+  const { dir, targetDocument } = useFluent_unstable();
   const expandIconRotation = treeItemState.open ? 90 : dir !== 'rtl' ? 0 : 180;
+  const groupperProps = useFocusableGroup();
+
+  const actionsRef = React.useRef<HTMLElement>(null);
+
+  const handleClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+    //  if click event originates from actions, ignore it
+    if (actionsRef.current && elementContains(actionsRef.current, event.target as Node)) {
+      return;
+    }
+    treeItemState.root.onClick?.(event);
+  });
 
-  // prevent default of a click from actions to ensure it doesn't open the treeitem
-  const handleActionsClick = useEventCallback((event: React.MouseEvent<HTMLElement>) => {
-    if (isResolvedShorthand(actions)) {
-      actions.onClick?.(event);
+  const handleKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === Enter) {
+      // if Enter keydown event comes from actions, ignore it
+      if (actionsRef.current && elementContains(actionsRef.current, event.target as Node)) {
+        return;
+      }
     }
-    event.preventDefault();
+    treeItemState.root.onKeyDown?.(event);
   });
 
+  const [keepActionsOpen, setKeepActionsOpen] = React.useState(false);
+
+  // Listens to focusout event on the document to ensure treeitem actions visibility on portal scenarios
+  // TODO: find a better way to ensure this behavior
+  React.useEffect(() => {
+    if (actionsRef.current) {
+      const handleFocusOut = (event: FocusEvent) => {
+        setKeepActionsOpen(elementContains(actionsRef.current, event.relatedTarget as Node));
+      };
+      targetDocument?.addEventListener('focusout', handleFocusOut, { passive: true });
+      return () => {
+        targetDocument?.removeEventListener('focusout', handleFocusOut);
+      };
+    }
+  }, [targetDocument]);
+
   return {
     ...treeItemState,
+    keepActionsOpen,
     components: {
       ...treeItemState.components,
       expandIcon: 'span',
       iconBefore: 'span',
       iconAfter: 'span',
       actions: 'span',
       badges: 'span',
+      groupper: 'span',
     },
+    root: {
+      ...treeItemState.root,
+      onClick: handleClick,
+      onKeyDown: handleKeyDown,
+    },
+    groupper: resolveShorthand(groupper, {
+      required: true,
+      defaultProps: groupperProps as ExtractSlotProps<Slot<'span'>>,
```

Comment:
let's also add `role="presentation"` so that screenreaders know this element is purely for layout